yarn-project/simulator/src/public/fixtures/public_tx_simulation_tester.ts

```diff
@@ -0,0 +1,170 @@
+import { MerkleTreeId, PublicExecutionRequest, type Tx } from '@aztec/circuit-types';
+import {
+  type AvmCircuitPublicInputs,
+  CallContext,
+  FunctionSelector,
+  GasFees,
+  GlobalVariables,
+  MAX_NOTE_HASHES_PER_TX,
+  MAX_NULLIFIERS_PER_TX,
+  NULLIFIER_SUBTREE_HEIGHT,
+  PUBLIC_DATA_TREE_HEIGHT,
+  PUBLIC_DISPATCH_SELECTOR,
+} from '@aztec/circuits.js';
+import { type ContractArtifact, encodeArguments } from '@aztec/foundation/abi';
+import { AztecAddress } from '@aztec/foundation/aztec-address';
+import { padArrayEnd } from '@aztec/foundation/collection';
+import { Fr } from '@aztec/foundation/fields';
+import { openTmpStore } from '@aztec/kv-store/lmdb';
+import { AvmTestContractArtifact } from '@aztec/noir-contracts.js/AvmTest';
+import { MerkleTrees } from '@aztec/world-state';
+
+import { BaseAvmSimulationTester } from '../../avm/fixtures/base_avm_simulation_tester.js';
+import { getContractFunctionArtifact, getFunctionSelector } from '../../avm/fixtures/index.js';
+import { SimpleContractDataSource } from '../../avm/fixtures/simple_contract_data_source.js';
+import { WorldStateDB } from '../public_db_sources.js';
+import { type PublicTxResult, PublicTxSimulator } from '../public_tx_simulator.js';
+import { createTxForPublicCalls } from './index.js';
+
+const TIMESTAMP = new Fr(99833);
+const DEFAULT_GAS_FEES = new GasFees(2, 3);
+
+const DISPATCH_SELECTOR = new FunctionSelector(PUBLIC_DISPATCH_SELECTOR);
+
+export type TestEnqueuedCall = {
+  fnName: string;
+  args: any[];
+  address?: AztecAddress;
+  isStaticCall?: boolean;
+  contractArtifact?: ContractArtifact;
+};
+
+export class PublicTxSimulationTester extends BaseAvmSimulationTester {
+  public static async create<T extends PublicTxSimulationTester>(
+    this: new (...args: any[]) => T,
+    skipContractDeployments = false,
+  ): Promise<T> {
+    const contractDataSource = new SimpleContractDataSource(skipContractDeployments);
+    const merkleTrees = await (await MerkleTrees.new(openTmpStore())).fork();
+    return new this(contractDataSource, merkleTrees, skipContractDeployments);
+  }
+
+  public async simulateTx(
```

Comment:
Just an idea (maybe for later): conceptually, IIRC, you should be able to create a TX (or calls) via aztecjs just like in the e2e tests. The calls to contract methods return a kind of call object, and I think you then should be able to use those instead of having to craft you own enqueuedcall/call structs.